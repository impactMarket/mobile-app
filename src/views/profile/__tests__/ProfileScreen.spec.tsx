import { assert } from 'chai';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import ProfileScreen from '../index';

const mockStore = configureMockStore();
const store = mockStore({});

describe('Profile screen test suite', () => {
    let screen: ShallowWrapper<any>;

    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('should render Profile screen correctly', () => {
        givenScreen();
        thenItRenderProperly();
    });

    function givenScreen() {
        assert.isDefined(ProfileScreen);

        screen = shallow(
            <Provider store={store}>
                <ProfileScreen />
            </Provider>
        );
    }

    function thenItRenderProperly() {
        expect(screen).toMatchSnapshot();
    }
});
