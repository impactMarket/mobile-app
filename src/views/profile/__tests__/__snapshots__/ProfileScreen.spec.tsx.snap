// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile screen test suite should render Profile screen correctly 1`] = `ShallowWrapper {}`;
